next/src/field/schema.ts

```diff
@@ -53,27 +101,50 @@ export function buildFieldSchema(
     return buildFieldObject(objectSchema, name, required)
   }
 
-  if (Array.isArray(schema.type)) {
+  if (schema.type === 'array') {
     throw new TypeError('Array type is not yet supported')
   }
 
-  const inputType = getInputType(schema)
+  const presentation = schema['x-jsf-presentation'] || {}
+  const errorMessage = schema['x-jsf-errorMessage'] || {}
 
-  const { title: _title, ...spreadSchema } = schema
+  // Get input type from presentation or fallback to schema type
+  const inputType = presentation.inputType || 'text'
 
+  // Build field without x-jsf prefixed properties
   const field: Field = {
-    ...spreadSchema,
-    ...schema['x-jsf-presentation'],
+    type: inputType, // Use presentation inputType for type
+    name,
     inputType,
-    type: inputType,
     jsonType: getJsonType(schema),
-    name,
     required,
+    isVisible: true, // Match main version default
+    computedAttributes: {}, // Match main version default
+    errorMessage,
   }
 
-  if (schema.title !== undefined) {
+  // Add optional fields if present
+  if (schema.title) {
     field.label = schema.title
   }
 
+  if (schema.description) {
+    field.description = schema.description
+  }
+
+  if (schema.maxLength) {
+    field.maxLength = schema.maxLength
+  }
+
+  if (schema.format) {
+    field.format = schema.format
+  }
```

Comment:
todo: These if's are not scalable, we'll endup with a LOT of them. We should spread all schema attributes into field by default [as explained here](https://linear.app/remote/issue/RMT-921)

Reply:
Agreed 👍